### PR TITLE
[SHELL] Enable to delete file type

### PR DESCRIPTION
**Pull request by katahiromz**
## Purpose

This PR implements "Delete" button of the "File Types" property sheet.

JIRA issue: [CORE-12906](https://jira.reactos.org/browse/CORE-12906)

## Proposed changes

- Add IDS_REMOVE_EXT resource string.
- Add DeleteExt function to delete the file type.
- Add IDC_FILETYPES_DELETE handler to FolderOptionsFileTypesDlg.

### CODE REVIEW

**Comment by katahiromz** [comment]
2k3:
2k3-delete

ros:
ros-delete

File: dll/win32/shell32/dialogs/folder_options.cpp [hunk 1]
```diff
@@ -1911,6 +1965,14 @@ FolderOptionsFileTypesDlg(
                         FileTypesDlg_AddExt(hwndDlg, newext.szExt, newext.szFileType);
                     }
                     break;
+                case IDC_FILETYPES_DELETE:
+                    LoadStringW(shell32_hInstance, IDS_REMOVE_EXT, Buffer, _countof(Buffer));
```
Comment (learn-more) [inline dll/win32/shell32/dialogs/folder_options.cpp]: This is a .cpp file, why not use CStringW?
`CStringW Buffer(MAKEINTRESOURCEW(IDS_REMOVE_EXT));`

File: dll/win32/shell32/dialogs/folder_options.cpp [hunk 2]
```diff
@@ -1749,6 +1774,13 @@ NewExtDlg_OnOK(HWND hwndDlg, NEWEXT_DIALOG *pNewExt)
             return FALSE;
         }
 
+        // Delete the extension
+        WCHAR szExt[20];
+        szExt[0] = L'.';
+        StringCchCopyW(&szExt[1], _countof(szExt) - 1, find.psz);
```
Comment (learn-more) [inline dll/win32/shell32/dialogs/folder_options.cpp]:
```
CStringW szExt;
szExt = find.psz;
```

**Comment by HBelusca** [comment]
@learn-more / @yagoulas : is it ok to use advapi32_vista in our shell32?

**Comment by HBelusca** [comment]
@katahiromz : Maybe it would be better to use, as an alternative to the vista+ RegDeleteKeyW, the [SHDeleteKeyW](https://msdn.microsoft.com/en-us/library/windows/desktop/bb773486(v=vs.85).aspx) function instead (exported by shlwapi.dll).
See these remarks too: https://stackoverflow.com/questions/33207470/what-is-the-difference-between-shdeletekey-and-regdeletetree .

File: dll/win32/shell32/dialogs/folder_options.cpp (hunk 1 above)
Comment (HBelusca) [reply dll/win32/shell32/dialogs/folder_options.cpp]: Note: using the ATL CStringW could have been done also in the previous patch that implemented the "create new file extension" dialog.

File: dll/win32/shell32/dialogs/folder_options.cpp (hunk 2 above)
Comment (katahiromz) [reply dll/win32/shell32/dialogs/folder_options.cpp]: OK.

File: dll/win32/shell32/dialogs/folder_options.cpp (hunk 1 above)
Comment (katahiromz) [reply dll/win32/shell32/dialogs/folder_options.cpp]: OK.

**Comment by katahiromz** [comment]
Used SHDeleteKeyW. Done.

**Comment by HBelusca** [comment]
Ok there is now a conflict with polish translation, could you please try to resolve it?

**Comment by katahiromz** [comment]
Done.

**Pull request merged**